### ISSUE TITLE
[BUG] vite plugin issue with react/tsx files in astro/Docs need to mention tsx files

### ISSUE THREAD

**Issue by hrmcdonald** [analogjs/analog#128]
### Please provide the environment you discovered this bug in.

When you setup an Astro project with MDX + React + Angular if your React components use typescript (`.tsx`) then pages that import them will fail to build. If the Angular plugin + components are removed everything works fine. If instead the React components are converted to `.jsx`  then everything will also work fine.

Also, awesome job with Analog, seems like a promising new development for the Angular community!

### Which area/package is the issue in?

vite-plugin-angular

### Description

This error is similar to #100 but seems to be a slightly different issue given we're using the latest version of the Vite Angular plugin here that should be ignoring `.tsx` files.

I have put together a recreation of the issue in stackblitz here: https://stackblitz.com/edit/withastro-astro-yrrypx?file=src%2Fpages%2Findex.mdx

Package versions installed:
- `@analogjs/astro-angular`: `0.1.0-beta.3`
- `@analogjs/vite-plugin-angular`: `0.2.0-alpha.15`
- `astro`: `1.6.2`

Something also seems to be off where the language server for the Angular component is ignoring `tsconfig.app.json` and only reports a valid file if you add certain typescript configurations to the base `tsconfig.json`. Not sure if this is related though.

### Please provide the exception or error you saw

```true
[ERROR] [plugin angular-compiler] File is missing from the TypeScript compilation.

    src/pages/react-test.tsx:0:0:
      0 │ 
        ╵ ^

  Ensure the file is part of the TypeScript program via the 'files' or 'include' property.

Build failed with 1 error:
src/pages/react-test.tsx:0:0: ERROR: [plugin: angular-compiler] File is missing from the TypeScript compilation.
Sourcemap for "/home/projects/withastro-astro-yrrypx/node_modules/@analogjs/astro-angular/src/server.js" points to missing source files
Angular is running in development mode. Call enableProdMode() to enable production mode.
 error   Unable to render / because it contains an undefined Component!
  Did you forget to import the component or is it possible there is a typo?
  File:
    /node_modules/astro/dist/runtime/server/jsx.js:34:15
  Stacktrace:
Error: Unable to render / because it contains an undefined Component!
Did you forget to import the component or is it possible there is a typo?
    at renderJSX (/node_modules/astro/dist/runtime/server/jsx.js:34:15)
    at eval (/node_modules/astro/dist/runtime/server/jsx.js:28:45)
    at Array.map (<anonymous>)
    at renderJSX (/node_modules/astro/dist/runtime/server/jsx.js:28:34)
    at renderJSX (/node_modules/astro/dist/runtime/server/jsx.js:38:16)
    at renderJSX (/node_modules/astro/dist/runtime/server/jsx.js:78:22)
    at async renderChild (/node_modules/astro/dist/runtime/server/render/any.js:10:11)
    at async Module.renderChild (/node_modules/astro/dist/runtime/server/render/any.js:23:5)
    at async Module.renderSlot (/node_modules/astro/dist/runtime/server/render/slot.js:19:22)
```

### Other information

<img width="972" alt="image" src="https://user-images.githubusercontent.com/39349270/199249383-66025b4c-af07-4267-97a0-acf4e0ca4f5d.png">

### I would be willing to submit a PR to fix this issue

- [X] Yes
- [ ] No

**Comment by brandonroberts** [comment]
The fix here is to add `"src/**/*.tsx"` to the `include` in the `tsconfig.app.json`. Should probably add that to the example `tsconfig.app.json` in the docs. Would you like to submit a PR?

**Comment by hrmcdonald** [comment]
Thanks for looking into this. That definitely fixes the issue in the stackblitz example I shared.

I did have this issue locally though despite already having added this configuration due to its mention in another issue thread. I believe the cause for the issue there turned out to be something with PNPMs default hoisting method, at least how our project was configured. Since adjusting that I've come back to this and am not seeing this issue again. It was either that or Vite was somehow caching enough to cause this error even after the config change previously.

Either way, I'd be happy to update the docs. I'll try to get a PR for that submitted in the next couple of days.
